### PR TITLE
add generated_data to repo to facilitate CI

### PR DESCRIPTION
**Pull request by peterboncz**
- Commit the generated data into the repo, because this facilitates running the tests in CI
  (since generating the data requires to equip the test runners with pyspark, which at least motherduck prefers not to).

- The data is minimized to 5MB, by setting the scale factor 10x smaller, and from expected_results/ only keeping last/. 
   This motivates rewriting some testcases that used expected_results/q08 into expected_results/last (which is the same)

- Removed one testcase, which tests if we get an error when parquet is not loaded (in MotherDuck, when we run iceberg in so-called "remote_only" mode, this will actually not raise an error).
   It was kind of a trivial test, so I hope this is OK..

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,11 +9,11 @@ include extension-ci-tools/makefiles/duckdb_extension.Makefile
 
 # Custom makefile targets
 data: data_clean
-	python3 scripts/test_data_generator/generate_iceberg.py 0.01 data/iceberg/generated_spec1_0_01 1
-	python3 scripts/test_data_generator/generate_iceberg.py 0.01 data/iceberg/generated_spec2_0_01 2
+	python3 scripts/test_data_generator/generate_iceberg.py 0.001 data/iceberg/generated_spec1_0_01 1
```
Comment (samansmink) [inline Makefile]: nit: the name of the directory is now no longer correct which could be a little confusing

**Comment by samansmink** [comment]
windows CI is failing due to some upstream issue, however the tests also seem to fail due to only the json metadata files being added and not the avro and parquet files

**Comment by peterboncz** [comment]
> windows CI is failing due to some upstream issue, however the tests also seem to fail due to only the json metadata files being added and not the avro and parquet files

apologies for that messup. I have made the name change and made sure all files are pushed now.

**Comment by samansmink** [comment]
Thanks for the changes! looks good!

**Pull request merged**